### PR TITLE
WIP: More resource cleanup

### PR DESCRIPTION
**Pull request by tas50**
We're not consistent here in a lot of the resources

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: lib/chef/resource/batch.rb [hunk 1]
```diff
@@ -21,7 +21,7 @@
 class Chef
   class Resource
     class Batch < Chef::Resource::WindowsScript
-
+      resource_name :batch
       provides :batch, os: "windows"
```
Comment (lamont-granquist) [inline lib/chef/resource/batch.rb]: can remove this line

File: lib/chef/resource/dsc_resource.rb [hunk 2]
```diff
@@ -20,6 +20,7 @@
 class Chef
   class Resource
     class DscResource < Chef::Resource
+      resource_name :dsc_resource
       provides :dsc_resource, os: "windows"
```
Comment (lamont-granquist) [inline lib/chef/resource/dsc_resource.rb]: also can be removed

File: lib/chef/resource/launchd.rb [hunk 3]
```diff
@@ -22,6 +22,7 @@
 class Chef
   class Resource
     class Launchd < Chef::Resource
+      resource_name :launchd
       provides :launchd, os: "darwin"
```
Comment (lamont-granquist) [inline lib/chef/resource/launchd.rb]: also delete

File: lib/chef/resource/macosx_service.rb [hunk 4]
```diff
@@ -21,37 +21,19 @@
 class Chef
   class Resource
     class MacosxService < Chef::Resource::Service
-
+      resource_name :macosx_service
       provides :macosx_service, os: "darwin"
```
Comment (lamont-granquist) [inline lib/chef/resource/macosx_service.rb]: delete the above line as well, the next line for `:service` is clearly correct though.

File: lib/chef/resource/powershell_package.rb [hunk 5]
```diff
@@ -23,17 +23,12 @@ class Resource
     class PowershellPackage < Chef::Resource::Package
       include Chef::Mixin::Uris
 
+      resource_name :powershell_package
       provides :powershell_package, os: "windows"
```
Comment (lamont-granquist) [inline lib/chef/resource/powershell_package.rb]: delete here as well

File: lib/chef/resource/windows_path.rb [hunk 6]
```diff
@@ -21,7 +21,7 @@
 class Chef
   class Resource
     class WindowsPath < Chef::Resource
-
+      resource_name :windows_path
       provides :windows_path, os: "windows"
```
Comment (lamont-granquist) [inline lib/chef/resource/windows_path.rb]: and here

File: lib/chef/resource/windows_task.rb [hunk 7]
```diff
@@ -21,7 +21,6 @@
 class Chef
   class Resource
     class WindowsTask < Chef::Resource
-
       resource_name :windows_task
       provides :windows_task, os: "windows"
```
Comment (lamont-granquist) [inline lib/chef/resource/windows_task.rb]: and here

**Comment by lamont-granquist** [comment]
you have some broken unit tests.

the ones i looked at were all checking to see if the provider was set where you've deleted `@provider =` lines out of the initializer.  you should delete those specs.  the code is bypassing the provider resolver and testing that the provider resolver is being bypassed, and both code + test are bad.

**Comment by tas50** [comment]
I'm actually going to break this into individual resource changes.

**Pull request closed without merging**